### ISSUE TITLE
[Desktop] Brave somewhat crashes after machine wakes up from suspend

### ISSUE THREAD

**Issue by kbsali** [brave/brave-browser#11678]
## Description 

Since v1.13.82 (I am now running v1.13.86) Brave "sort of" crashes when waking up my computer after a suspend (on my laptop, when opening the lid after closing it :)).
I use "sort of crashes" because the windows are still opened but they all show a black screen and I can only see part of the tab but no other button/ui element what so ever (see screenshot below).
I am running Ubuntu Mate 20.04, but a colleague of mine has the exact same issue on Elementary 5.1.7 (based on Ubuntu 18.04).
Killing & restarting brave "fixes" the issue.

## Steps to Reproduce

   1. suspend Linux (close your laptop lid)
   2. wake-up (open your laptop lid)

## Actual result:

![2020-09-11_14-42](https://user-images.githubusercontent.com/53676/92926997-32c25380-f43d-11ea-875d-5a2b1d56deb0.png)

## Expected result:

Brave should run as usual! :sweat_smile: 

## Reproduces how often: 

Always

## Brave version (brave://version info)

(also happening on 1.13.82)

Brave | 1.13.86 Chromium: 85.0.4183.102 (Official Build) (64-bit)

-- | --
Revision | ffe848af6a5df4fa127e2929331116b7f9f1cb30-refs/branch-heads/4183@{#1770}
OS | Linux
JavaScript | V8 8.5.210.20
Flash | (Disabled)
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.102 Safari/537.36
Command Line | /opt/brave.com/brave/brave --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --enable-features=AutoupgradeMixedContent,PassiveMixedContentWarning,DnsOverHttps,OmniboxContextMenuShowFullUrls,MixedContentSiteSetting,WebUIDarkMode,PasswordImport,ReducedReferrerGranularity,PrefetchPrivacyChanges --disable-features=NotificationTriggers,TextFragmentAnchor,SafeBrowsingEnhancedProtection,AutofillEnableAccountWalletStorage,AllowPopupsDuringPageUnload,TabHoverCards,SmsReceiver,PrivacySettingsRedesign,VideoPlaybackQuality,AutofillServerCommunication,PasswordCheck --flag-switches-begin --load-media-router-component-extension=1 --flag-switches-end
Executable Path | /opt/brave.com/brave/brave
Profile Path | /home/kevin/.config/BraveSoftware/Brave-Browser/Profile 1

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
	- yes
- Can you reproduce this issue with the beta channel? 
	- yes (Brave Browser 85.1.15.46 beta)
- Can you reproduce this issue with the nightly channel? 
	- **YES** (Chromium 85.0.4183.102 snap) !!!

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 
	- **YES** (Chromium 85.0.4183.102 snap) !!!

## Miscellaneous Information:

Just checked with the same colleague of mine, he has Chromium v84 and it works as expected.
So the issue seems to come from the version of Chromium Brave is based on, right?

Thanks for you help

**Comment by kbsali** [comment]
see related bug https://bugs.launchpad.net/ubuntu/+source/chromium-browser/+bug/1894432

**Comment by howdev** [comment]
I can confirm this on V1.14.84 with chromium 85.  It is not crash as described. Is actually corrupted display. 
Corrupted display with black screen in the browser after hibernate or suspend. 

I found the fix  is to enable Vulkan flag. The problem is gone.

**Comment by howdev** [comment]
But the Vulkan flag causes browser to lag

**Comment by howdev** [comment]
I tried the new release Brave 1.15.72 chromium 86. still same problem. I tried Beta channel also same problem
When you end the Brave gpu process, it restarts itself, then the UI returns to normal

**Comment by howdev** [comment]
I have a workaround, thanks to Linux ability. 
use hibernate, suspend hook
write a script in here /lib/systemd/system-sleep 
it is important to us HUP otherwise the gpu process won't work
#!/bin/sh

case "$1" in
	post)
		pkill -HUP -f type=gpu
	;;
esac

**Comment by jaubin** [comment]
Same there, but it worsens with Brave 86 at least with KDE : 
- If the window is maxed out before suspending, after resuming transparency effects are completely broken on the whole desktop, and the only workaround is to log out/log in.

I see it occuring with NVidia drivers, but it seems that when using Mesa drivers the problem does not appear.

**Comment by howdev** [comment]
@jaubin Use systemd hibernate suspend hook. works fine with it. The issue seemed to do with Chromium as it renders the whole UI in its own  toolkit and not Gtk or Qt.

**Comment by kbsali** [comment]
As far as I'm concerned, it's been working correctly ever since I updated the menu command to execute : `/usr/bin/brave-browser-stable --use-gl=desktop %U`

**Comment by howdev** [comment]
@kbsali brave browser does not have any commandline options

**Comment by kbsali** [comment]
does it not "inherit" the options of chromium

**Comment by howdev** [comment]
Command line is not related to the engine. you cannot use -help or ---help with brave, means there is no command options.

**Comment by kbsali** [comment]
No offense, but you can pass parameters to the brave executable, see `man brave-browser-stable` for the available options (and funny enough, the options are exactly the same as for chromium `chromium --help`).
Some of the options are not documented here though (`--use-gl` being one for them).

**Comment by howdev** [comment]
@kbsali  that is strange implementation without listing in  ---help. ---use-gl does work but not convenient to launch Brave. Someone suggested there is conf to put the flags. or you can find the same option in the chrome://flags. Either way, as long there is a workaround

**Comment by rebron** [comment]
possibly related to: https://bugs.chromium.org/p/chromium/issues/detail?id=791913

**Comment by JohnGalt1717** [comment]
I'm also getting this on Ubuntu 20.04 with Nvidia graphics and latest proprietary drivers.

Chrome on same machine doesn't have it.

**Comment by howdev** [comment]
@JohnGalt1717  That is strange.  It is not just the browser window that is corrupted, it is the whole UI. So it should happen on Chrome too.  Everything, the rendering and UI are done by Chromium

**Comment by jaubin** [comment]
Yes and when you switch from tab to tab you can see that actually the tabs become some kind of "transparent" on white backgrounds.
    Le mardi 20 octobre 2020 à 02:01:15 UTC+2, howdev <notifications@github.com> a écrit :  
 
 


@JohnGalt1717 That is strange. It is not just the browser window that is corrupted, it is the whole UI. So it should happen on Chrome too. Everything, the rendering and UI are done by Chromium

**Comment by JohnGalt1717** [comment]
For me, the page itself renders fine after a while (parts missing at first but a simple scroll fixes it). It's just the chrome on the outside.  Edge dev on linux also works fine recovering from standby.

**Comment by ripekern** [comment]
I was having this difficulty and found the solution is go to settings in the brave-browser and disable hardware-acceleration. [ settings > system > Use hardware acceleration when available > OFF ]

**Comment by GabeVilela** [comment]
I had this same issue early today. After waking up my PC, I started watching a YouTube video, then the web view blinked.
Later, I maximized another Brave window (which I had minimized), and it appeared all black with a few parts rendering normally only while resizing it.
Restarting the browser fixed the problem.

BTW, I sent a crash report with the ID `1ee24700-8ce9-8106-0000-000000000000`
I'm not sure if it has some correlation with this issue or not, but the timestamp does match with the time frame of when the problem happened.

**More info about my system**
- Brave 1.26.67 Chromium: 91.0.4472.114 64 bits
- OS: Windows 10 OS Version 2009 (Build 19043.1052)
- GPU: Nvidia GeForce GTX 960
- GPU Driver: 466.27

**Comment by codeonblocks** [comment]
I'm also having this issue.

**Comment by bsclifton** [comment]
Hi folks - I am going to close this issue as it's a bit old without an update

I did put together some notes and a [new process for when folks are having GPU issues](https://github.com/brave/brave-browser/wiki/GPU-issues---incident-reports).

If you're reading this and you are having GPU issues, you can help me by creating a new issue here describing the problem on GitHub and then attaching the report from brave://gpu.
